theano/sandbox/multinomial.py

```diff
@@ -215,6 +215,146 @@ def make_node(self, pvals, unis, n=1):
         out = T.tensor(dtype=odtype, broadcastable=pvals.type.broadcastable)
         return Apply(self, [pvals, unis, as_scalar(n)], [out])
 
+    def c_code_cache_version(self):
+        return None
+
+    def c_code(self, node, name, ins, outs, sub):
+        (pvals, unis, n) = ins
+        (z,) = outs
+        if self.odtype == 'auto':
+            t = "NPY_INT64"
+        else:
+            t = theano.scalar.Scalar(self.odtype).dtype_specs()[1]
+            if t.startswith('theano_complex'):
+                t = t.replace('theano_complex', 'NPY_COMPLEX')
+            else:
+                t = t.upper()
+        fail = sub['fail']
+        return """
+        // create a copy of pvals matrix
+        PyArrayObject* pvals_copy = NULL;
+
+        if (PyArray_NDIM(%(pvals)s) != 2)
+        {
+            PyErr_Format(PyExc_TypeError, "pvals wrong rank");
```

Comment:
I understand you took hit from MultinomialFromUniform but I think the error message from make_node is more explicit. If you agree could you please use 'pvals ndim should be 2' and modify it in MultinomialFromUniform.c_code at the same time.